src/node_worker.cc

```diff
@@ -274,10 +318,23 @@ void Worker::Run() {
     if (thread_stopper_.IsStopped()) return;
     {
       HandleScope handle_scope(isolate_);
-      Local<Context> context = NewContext(isolate_);
+      Local<Context> context;
+      {
+        // We create the Context object before we have an Environment* in place
+        // that we could use for error handling. If creation fails due to
+        // resource constraints, we need something in place to handle it,
+        // though.
+        TryCatch try_catch(isolate_);
+        context = NewContext(isolate_);
+        if (context.IsEmpty()) {
+          // TODO(addaleax): Inform the target about the actual underlying
+          // failure.
+          custom_error_ = "ERR_WORKER_OUT_OF_MEMORY";
```

Comment:
Why is it set to `ERR_WORKER_OUT_OF_MEMORY` here? Won't the NearHeapLimit callback handle that already?

Reply:
Because we don’t really know what happened. The only time I’ve seen this fail in practice is due to running out of stack space, not out of memory, but either way it’s rare enough that I think “out of memory” is good enough.